packages/wheelhouse-frontend/src/components/environment-variables/EnvironmentVariablesDisplay.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import Sidebar from "../reusables/Sidebar";
+import "./EnvironmentVariablesDisplay.css";
+import { DEVELOPMENT_ENV_CHANGE } from "wheelhouse-core";
+
+class EnvironmentVariablesDisplay extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      current: "",
+    };
+  }
+
+  changeCheckmark(checked) {
+    this.setState({checked});
+  }
+
+  buttonClick(name, selectedValue) {
+    this.setState({
+      current: selectedValue
+    });
+    this.props.dispatch({
+      type: DEVELOPMENT_ENV_CHANGE,
+      variableName: name,
+      currentValue: selectedValue
+    });
+  }
+
+  inputChange(name, event) {
+    this.setState({
+      current: event.currentTarget.value
+    });
+    this.props.dispatch({
+      type: DEVELOPMENT_ENV_CHANGE,
+      variableName: name,
+      currentValue: event.currentTarget.value
+    });
+  }
+
+  render() {
+    const { env } = this.props;
+    const variable = Object.keys(env).map((name) => {
+      const data = env[name];
+      const presetValues = data.presetValues;
+      const newInput = <input
+        className="customInput"
+        type="text"
+        placeholder="custom"
+        onKeyUp={this.inputChange.bind(this, name)}>
+      </input>;
+
+      const reformattedArray = presetValues.map((obj) => {
+        let button = obj.value;
+        let checked;
+        if (button === this.state.current) {
```

Comment:
The checkbox can't be based on local state -- it needs to be visible across refreshes and browser changes. But I think that's easy enough to do if you just change this if statement to 

```
if (button === this.props.env[name].currentValue) {
```
That way it's based on the actual current value, rather than the last thing the user checked